### PR TITLE
feature/jenkins 35828 - pipeline card

### PR DESCRIPTION
**Pull request by cliffmeyers**
Related to issue #JENKINS-35828 . 

Summary of this pull request: 
* New Pipeline Card component
* This one has a few small design iterations to go through but the code changes should be very minor.
* **Needs merge and publish of JDL for code in jenkinsci/jenkins-design-language#68  and jenkinsci/jenkins-design-language#73**

@reviewbybees

### CODE REVIEW

File: blueocean-personalization/src/main/js/components/PipelineCard.jsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+        return (
+            <div className={`pipeline-card ${bgClass}`}>
+                <LiveStatusIndicator result={this.props.status} noBackground />
+
+                <span className="name">
+                    {this.props.organization} / {this.props.pipeline}
+                </span>
+
+                { this.props.branch &&
```
Comment (kzantow) [inline blueocean-personalization/src/main/js/components/PipelineCard.jsx]: I like this style very much, didn't think of using it myself :)

**Comment by kzantow** [comment]
:bee: AFAICT

File: blueocean-personalization/package.json [hunk 2]
```diff
@@ -26,6 +26,7 @@
     "enzyme": "^2.2.0",
     "eslint": "2.8.0",
     "eslint-plugin-react": "^5.0.1",
+    "eslint-to-editorconfig": "1.2.0",
```
Comment (tfennelly) [inline blueocean-personalization/package.json]: Cool :)

Comment (tfennelly) [inline blueocean-personalization/package.json]: Runs via `fixLint` I assume?

File: blueocean-personalization/src/main/js/components/PipelineCard.jsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+ * onRunClick: callback invoked when 'Run Again' is clicked
+ * onFavoriteToggle: callback invokved when favorite checkbox is toggled.
+ */
+export class PipelineCard extends Component {
+
+    static _getBackgroundClass(status) {
+        return status !== null && status.length > 0 ?
+            `${status.toLowerCase()}-bg-lite` :
+            '';
+    }
```
Comment (tfennelly) [inline blueocean-personalization/src/main/js/components/PipelineCard.jsx]: Is this any different to ...

```javascript
PipelineCard._getBackgroundClass = function(status) {
    // etc...
};
```

I'd assume it is. Just wondering.

**Comment by tfennelly** [comment]
LGTM 🐝

File: blueocean-personalization/package.json (hunk 2 above)
Comment (cliffmeyers) [inline blueocean-personalization/package.json]: This only generates the editorconfig unfortunately, I just added it to make sure the two were in sync with each other. It more or less matches the one from dashboard fine.

File: blueocean-personalization/src/main/js/components/PipelineCard.jsx (hunk 3 above)
Comment (cliffmeyers) [inline blueocean-personalization/src/main/js/components/PipelineCard.jsx]: Yep, same thing.

**Comment by i386** [comment]
@cliffmeyers 🐛 the favourite stars are white (outlined/solid states) for all card statuses. We should also have a fix for IE11.

blueocean sketch 2016-07-04 09-54-09

**Comment by michaelneale** [comment]
Looks good to me, modulo the white-star for visibility as mentioned by @i386. 🐝  once that is taken care of.

File: blueocean-personalization/src/main/js/components/PipelineCard.jsx (hunk 3 above)
Comment (michaelneale) [inline blueocean-personalization/src/main/js/components/PipelineCard.jsx]: Good to know...

**Comment by michaelneale** [comment]
even in this state, is it worth having some unit tests?

**Comment by michaelneale** [comment]
Have corrected (via @i386) the star colour. 

@i386 was going to provide a sprite sheet so it can work in IE without needing CSS filters.

File: blueocean-personalization/src/main/less/components/pipeline-card.less [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+    .progress-spinner.not_built {
+        circle {
+            stroke: white;
+        }
+    }
+
+    // force "favorite" checkbox to white when on unstable card
+    // FIXME: doesn't work in IE11
+    .checkbox {
```
Comment (i386) [inline blueocean-personalization/src/main/less/components/pipeline-card.less]: Avoid this filter method by using this specially crafted white star sprite sheet I prepared :) 

https://www.dropbox.com/s/3chchxbw4gpjjy5/stars-sprits-card.svg?dl=0

**Comment by i386** [comment]
LGTM. Please use the sprite sheet I provided instead of the filter method for inverting the star color (IE11 won't support filters anyway)

**Comment by cliffmeyers** [comment]
@i386 thanks for the updated SVG, it's now been integrated.

@michaelneale also added a few unit tests for the card - thanks for keeping me honest.

**Comment by i386** [comment]
@cliffmeyers nice! Thanks mate - this LGTM.

**Comment by michaelneale** [comment]
🐝

**Pull request merged**